### DISCUSSION TITLE
Downloading 'Custom' FHIR resources via DownloadManager

### DISCUSSION THREAD

**Discussion by ellykits** [google/android-fhir#2625] [Q&A]
Currently, the SDK defines this `typealias`:

`typealias ResourceSearchParams = Map<ResourceType, ParamMap>`

This type is used by the `ResourceParamsBasedDownloadWorkManager` to build request URLs used by the download manager to download resources. Would it be possible to modify this to support downloading of 'Custom' resource types not part of the `ResourceType` enum by changing the map's key to type `String`?

`typealias ResourceSearchParams = Map<String, ParamMap>`

Any other suggestion to support downloading custom resources is welcomed. For context, we would want to download a custom resource (`LocationHierarchy`) in FHIRCore using the DownloadManager.

**Comment by ndegwamartin** [comment]
FHIR Allows the spec to be extended, mean we could potentially have HAPI FHIR Structures that are not in the existing concrete implementations. This would solve for any such extensions.

**Comment by aditya-07** [comment]
@ellykits I agree that we should update `ResourceSearchParams` to allow downloading of custom structures since request URLs are build using the provided resource type.

There might be other changes required in the Database and Search api to effectively handle extended / custom resources since they are effectively handled in the database layer as the base fhir resource they extend e.g.  `LocationHierarchy` as `Bundle`.

**Comment by ellykits** [comment]
@aditya-07 Is there any plan to start working on this? We implemented a `CustomSyncWorker` as a workaround and we are currently missing out on some functionalities including: 

1. Calculating sync progress (in percentage),
2. Getting sync status notifications.
3. Pagination of requests etc
